src/vcpkg/archives.cpp

```diff
@@ -152,6 +189,10 @@ namespace
         {
             extract_tar_cmake(paths.get_tool_exe(Tools::CMAKE), archive, to_path);
         }
+        else if (Strings::case_insensitive_ascii_equals(ext, ".exe") && Strings::starts_with(filename, "7z"))
```

Comment:
I would like to see whatever is calling this call win32_extract_7z_installer directly instead of trying to reverse engineer what the right thing is based on the file name.

Reply:
yeah I understand that. The problem is I found no good hook into it. tooldata seems to only know about archives and changing that is a bigger endeavor and i did not see a general solution. The extracting call is at:
https://github.com/microsoft/vcpkg-tool/blob/42152a63f8bd79d07cd2271c7c82fe67e7f20300/src/vcpkg/tools.cpp#L207-L211

Reply:
How do you feel about https://github.com/BillyONeal/vcpkg-tool/commit/60f2137e47f273e19f4b6366a74cddf730d9c2c6 ?

Reply:
> How do you feel about https://github.com/BillyONeal/vcpkg-tool/commit/60f2137e47f273e19f4b6366a74cddf730d9c2c6 ?

adding a new tool provider looked like the old way of adding tools (probably because it does not scale well). I was more thinking along the line to add an `<extractCommand>` field and an `<extractEnvironment>` field to the xml.  `<extractCommand>`  would have placeholders for positional arguments. So `{1}` would always be the downloaded archive and `{2}` would be the target directory. Basically just using fmt to build the cmd line.

Furthermore if you don't want to switch on the file extensions then maybe add a field `<extractMethod>`  to the xml?

Reply:
>adding a new tool provider looked like the old way of adding tools (probably because it does not scale well).

No, it's the normal way of adding tools. The XML file only existed because way back when we also needed to be able to fetch that stuff from bash to bootstrap where we could only parse with regex.

In particular, the xml file has no way to describe how to fetch version information from an already installed copy of anything.

If anything the XML file should be outright deleted now that we never consult it from bash.

Reply:
ok. Hmm, I thought that the xml is/was a good approach to have data/code separation. Don't mind if it gets changed to json instead ;) . As I said the current ToolProvider doesn't really scale and the amount of boilerplate to setup a new one is significant. Even worse the big if in `tools.cpp` with all the different cases handled all slightly different but somehow similar.